### PR TITLE
Implement Base Testing Framework for Python Application

### PR DESCRIPTION
**Pull request by sajosam**
This pull request establishes a foundational testing framework for the Python application to enhance testing capabilities for future development.

**Changes Made:**
Created a basic structure to support:
1. Integration Testing
2. Functional Testing
3. Unit Testing

### CODE REVIEW

File: app/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+from typing import Generator, Any
```
Comment (agberoz) [inline app/tests/conftest.py:1]: can you move `tests` folder to base directory

File: app/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import pytest
```
Comment (agberoz) [inline app/tests/conftest.py:2]: Add documentation for testing instructions in Readme.md

**Review by agberoz — CHANGES_REQUESTED** [review]

**Comment by sajosam** [comment]
@agberoz 
- Added test file to root directory
- Added readme file for testing
image
image

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+## Getting Started
+
+This project includes functional, integration, and unit testing using pytest. Tests are organized into separate folders based on their type, and a conftest.py file is used to manage shared configurations and fixtures.
+
+### Project Structure
+- `functional/`: Contains functional tests (`test_*.py`).
+- `integration/`: Contains integration tests (`test_*.py`).
+- `unittest/`: Contains unit tests (`test_*.py`).
+
+## Requirements
```
Comment (agberoz) [inline tests/README.md:10]: add dependency with version to requirement.txt

**Review by agberoz — APPROVED** [review]
LGTM

**Pull request merged**